sympy/simplify/fu.py

```diff
@@ -1049,7 +1049,7 @@ def f(rv):
                     arg_num = fi.args[0]
                     for arg_denom in list(arg_denoms):
                         base = arg_num/arg_denom
-                        if fi.func != arg_denoms[arg_denom]:
+                        if not (fi.func is cos and arg_denoms[arg_denom] is cos) :
```

Comment:
```suggestion
                        if not (isinstance(fi, cos) for arg_denoms[arg_denom] is cos) :
```

Reply:
I am not sure how to use the `is_instance` with the `cos` of `arg_denoms` part since `isinstance(cos, cos)` is False.

Reply:
I have used `==` operator in this case in the updated code. Should I use `isinstance` instead?